tests/server_tests.js

```diff
@@ -400,6 +404,33 @@ describe('Server', function() {
       expect(config.get('port')).to.eq(server.server.address().port);
     });
   });
+
+  describe('injectMiddleware', function() {
+    it('passes app and server', function(done) {
+      config = new Config('dev', {
+        port: 0,
+        cwd: 'tests',
+        middlewear: [
+          function(app, server) {
+            expect(app).to.be.instanceof(express);
+            expect(server).to.be.instanceof(BaseServer);
+            done();
+          }
+        ]
+      });
+      server = new Server(config);
+      server.once('server-start', function() {
+        done();
```

Comment:
Yes, but the `done` function should generally only be called once in a test. In your test you call it twice, ones after the server is booted and once during the middleware creation. I guess the middleware one should be dropped. Could you make sure that it no longer passes when you stop passing `app` in the code?